src/biogeochem/Fan2CTSMMod.F90

```diff
@@ -0,0 +1,1059 @@
+    real(r8), intent(out) :: tan_manure_spread_col(bounds%begc:bounds%endc) ! output, calculated from the above and stored manure
+    ! N excreted by animals allocated to mixed production systems temporarily grazing on grasslands:
+    real(r8), intent(inout) :: n_manure_graze_col(bounds%begc:bounds%endc)
+    ! N excreted by animals in mixed systems, total
+    real(r8), intent(out) :: n_manure_mixed_col(bounds%begc:bounds%endc)
+    ! NH3 emission fluxes from manure storage and housings, gN/m2/s
+    real(r8), intent(out) :: nh3_flux_stores(bounds%begc:bounds%endc), nh3_flux_barns(bounds%begc:bounds%endc)
+    ! total nitrogen flux transferred out of a crop column (manure spreading + temporary grazing)
+    real(r8), intent(out) :: man_n_transf(bounds%begc:bounds%endc)
+    ! Total nitrogen excreted in barns
+    real(r8), intent(out) :: man_n_barns(bounds%begc:bounds%endc)
+    ! fraction of manure excreted when grazing
+    real(r8), intent(out) :: grz_fract(bounds%begc:bounds%endc)
+    ! TAN fraction in excreted N
+    real(r8), intent(in) :: tan_fract_excr
+    integer                  , intent(in)    :: num_soilc       ! number of soil columns in filter
+    integer                  , intent(in)    :: filter_soilc(:) ! filter for soil columns
+
+    integer :: begg, endg, g, l, c, il, counter, col_grass, status, p
+    real(r8) :: flux_avail_rum, flux_avail_mg, flux_grazing, invscale
+    real(r8) :: tempr_ave, windspeed_ave ! windspeed and temperature averaged over agricultural patches
+    real(r8) :: tempr_barns, tempr_stores, vent_barns, flux_grass_crop, tempr_min_10day, &
+         flux_grass_graze, flux_grass_spread, flux_grass_spread_tan, flux_grass_crop_tan
+    real(r8) :: cumflux, totalinput, total_to_store, total_to_store_tan
+    ! dimensions are (type of flux, ruminant/other)
+    real(r8) :: fluxes_nitr(num_fluxes,2), fluxes_tan(num_fluxes,2) 
+    ! The fraction of manure applied continuously on grasslands (if present in the gridcell)
+    real(r8), parameter :: kg_to_g = 1e3_r8
+    logical :: is_grass
+    
+    begg = bounds%begg; endg = bounds%endg
+    nh3_flux_stores(bounds%begc:bounds%endc) = 0_r8
+    nh3_flux_barns(bounds%begc:bounds%endc) = 0_r8
+    man_n_barns(bounds%begc:bounds%endc) = 0.0_r8
+    
+    totalinput = 0.0
+    cumflux = 0.0
+    
+    do g = begg, endg
+       ! First find out if there are grasslands in this cell. If yes, a fraction of
+       ! manure can be diverted to them before storage.
+       col_grass = ispval
+       do il = 1, max_lunit
+          l = grc%landunit_indices(il, g)
+          if (l == ispval) cycle
+          if (lun%itype(l) == istsoil) then
+             do p = lun%patchi(l), lun%patchf(l)
+                is_grass = patch%itype(p) == nc4_grass &
+                     .or. patch%itype(p) == nc3_nonarctic_grass &
+                     .or. patch%itype(p) == nc3_arctic_grass
+                if (is_grass .and. col%wtgcell(patch%column(p)) > 1e-6) then
```

Comment:
More checks of subgrid weights in this subroutine, which should be avoided.

Reply:
One particular reason why we need to avoid checks of subgrid weights is: We sometimes need to treat 0-weight patches or columns as being active. These 0-weight patches or columns should behave exactly the same as non-zero-weight patches or columns. It looks like this requirement is violated in a number of places in this code.

Reply:
Checks of subgrid weight have been removed from this routine - thanks. There are still multiplications and divisions by subgrid weights. These may be needed for now as with an earlier comment. I'm not entirely sure how it will work to have these multiplications and divisions without the checks that are now gone, but if it's working for you, then I'll mark this as resolved for now and we can come back to this along with the bigger rework of the subgrid stuff in FAN, as noted elsewhere.